### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by mrBliss**
Highlights:

* https://github.com/input-output-hk/cardano-ledger-specs/pull/1915
* https://github.com/input-output-hk/cardano-ledger-specs/pull/1922
* https://github.com/input-output-hk/cardano-ledger-specs/pull/1902
* https://github.com/input-output-hk/cardano-ledger-specs/pull/1923
* https://github.com/input-output-hk/cardano-ledger-specs/pull/1927
* https://github.com/input-output-hk/cardano-ledger-specs/pull/1929

### CODE REVIEW

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs [hunk 1]
```diff
@@ -541,6 +555,83 @@ translateLedgerViewByronToShelleyWrapper =
...
+{-------------------------------------------------------------------------------
+  Translation from one Shelley-based era to another Shelley-based era
+-------------------------------------------------------------------------------}
+
+instance ( ShelleyBasedEra era
+         , ShelleyBasedEra (SL.PreviousEra era)
+         , EraCrypto (SL.PreviousEra era) ~ EraCrypto era
+         ) => SL.TranslateEra era ShelleyTip where
+  translateEra _ (ShelleyTip sno bno (ShelleyHash hash)) =
+      return $ ShelleyTip sno bno (ShelleyHash hash)
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs:567]: Use default instance based on `coerce`.

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs [hunk 2]
```diff
@@ -541,6 +555,83 @@ translateLedgerViewByronToShelleyWrapper =
...
+          shelleyLedgerTip        = tip'
+        , shelleyLedgerState      = state'
+        , shelleyLedgerTransition = ShelleyTransitionInfo 0
+        }
+
+instance ( ShelleyBasedEra era
+         , SL.TranslateEra era SL.Tx
+         , SL.TranslationError era SL.Tx ~ ()
+         ) => SL.TranslateEra era (GenTx :.: ShelleyBlock) where
+  type TranslationError era (GenTx :.: ShelleyBlock) = ()
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs:587]: Just forward it and ignore the error in the injection.

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs [hunk 3]
```diff
@@ -541,6 +555,83 @@ translateLedgerViewByronToShelleyWrapper =
...
+  translateEra ctxt (Comp (ShelleyTx _txId tx)) =
+    -- TODO will the txId stay the same? If so, we could avoid recomputing it
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs:589]: Ask the ledger.

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs [hunk 4]
```diff
@@ -563,11 +654,12 @@ translateChainDepStateShelleyToAllegraWrapper ::
        (ShelleyBlock (AllegraEra c))
 translateChainDepStateShelleyToAllegraWrapper =
     ignoringBoth $
-      Translate $ \_epochNo (WrapChainDepState stateShelley) ->
-        WrapChainDepState stateShelley
+      Translate $ \_epochNo (WrapChainDepState chainDepState) ->
+        -- Same protocol, same 'ChainDepState'
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs:658]: Comment: we don't have to apply any epoch-related changes, this happens when ticking, as usual.

File: ouroboros-consensus-cardano/ouroboros-consensus-cardano.cabal [hunk 5]
```diff
@@ -49,6 +49,7 @@ library
                      , cardano-prelude
                      , cardano-slotting
                      , shelley-spec-ledger
+                     , cardano-ledger-shelley-ma
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/ouroboros-consensus-cardano.cabal:52]: Don't need it yet

File: ouroboros-consensus-cardano-test/src/Test/ThreadNet/TxGen/Cardano.hs [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
-{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE LambdaCase           #-}
+{-# LANGUAGE NamedFieldPuns       #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE UndecidableInstances #-}
```
Comment (mrBliss) [inline ouroboros-consensus-cardano-test/src/Test/ThreadNet/TxGen/Cardano.hs:9]: Needed?

**Review by edsko — APPROVED** [review]
Reviewed over meet.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs [hunk 7]
```diff
@@ -16,10 +16,12 @@
 {-# LANGUAGE StandaloneDeriving         #-}
 {-# LANGUAGE TypeApplications           #-}
 {-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UndecidableInstances       #-}
```
[the remark this answers is not in the document]
Comment (mrBliss) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs:19]: Need for the `OtherHeaderEnvelopeError` type instance.

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs (hunk 1 above)
Comment (mrBliss) [reply ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs:567]: Doesn't work :slightly_frowning_face:

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs (hunk 3 above)
Comment (mrBliss) [reply ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs:589]: Asked.

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs (hunk 4 above)
Comment (mrBliss) [reply ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs:658]: I also realised that we don't need specific translations for `ChainDepState` and `LedgerView`.

File: ouroboros-consensus-cardano/ouroboros-consensus-cardano.cabal (hunk 5 above)
Comment (mrBliss) [reply ouroboros-consensus-cardano/ouroboros-consensus-cardano.cabal:52]: I do, for the `TranslateEra` instances.

File: ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs [hunk 8]
```diff
@@ -437,9 +439,9 @@ exampleNewEpochState = SL.NewEpochState {
     rewardUpdate :: SL.RewardUpdate era
     rewardUpdate = SL.RewardUpdate {
           deltaT    = SL.Coin 10
-        , deltaR    = SL.Coin 100
+        , deltaR    = SL.Coin 0
```
[the remark this answers is not in the document]
Comment (mrBliss) [reply ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs:442]: https://github.com/input-output-hk/cardano-ledger-specs/pull/1934#discussion_r510142825

File: ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs [hunk 9]
```diff
@@ -437,9 +439,9 @@ exampleNewEpochState = SL.NewEpochState {
     rewardUpdate :: SL.RewardUpdate era
     rewardUpdate = SL.RewardUpdate {
           deltaT    = SL.Coin 10
-        , deltaR    = SL.Coin 100
```
[the remark this answers is not in the document]
Comment (mrBliss) [reply ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs:440]: https://github.com/input-output-hk/cardano-ledger-specs/blob/1a2d7717682f8191cf818362df28ac20fac19b83/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs#L384-L401
And according to the spec (11.9), `deltaR` and `deltaF` are always negative.

**Comment by mrBliss** [comment]
bors merge

**Comment by mrBliss** [comment]
C'mon bors, why do you have to be like this? :slightly_frowning_face:

**Comment by edsko** [comment]
Should I be worried that you're talking to a bot?

**Comment by mrBliss** [comment]
> Should I be worried that you're talking to a bot?

You were AFK ... :smile:

**Comment by mrBliss** [comment]
bors ping

**Pull request merged**